### PR TITLE
Dynamic config

### PR DESCRIPTION
**Pull request by AvaniMakwana**
Authored-by: AvaniMakwana
One of the multiple PRs that will resolve #1178

Description
Integrating Elide Dynamic Configuration changes.
Added dynamic config hjson files to resources.
Additional tables added to changelog liquibase config
Additional test cases added for in memory compilation
This code depends on elide-5.x snapshot jars.
Motivation and Context
Please refer #1178

How Has This Been Tested?
Unit tests included.

License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/main/java/example/Settings.java [hunk 1]
```diff
@@ -65,28 +66,26 @@ public int getPort() {
...
+    public boolean enableSwagger() {
+        return true;
     }
 
     @Override
     public String getModelPackageName() {
-
         //This needs to be changed to the package where your models live.
         return "example.models";
+//    	return "dynamicconfig.models";
```
Comment (moizarafat) [inline src/main/java/example/Settings.java:77]: please remove commented line.

File: src/main/java/example/Settings.java [hunk 2]
```diff
@@ -65,28 +66,26 @@ public int getPort() {
...
+    }
+    
+    @Override
+    public boolean enableDynamicModelConfig() {
+        return true;
+    }
+    
+    @Override
+    public String getDynamicConfigPath() {
+        File file = new File("src/main/resources/models");
```
Comment (moizarafat) [inline src/main/java/example/Settings.java:87]: do we need to use File and absolute path? in the standalone code unit test  we just return `src/test/resources/models`

File: src/main/java/example/models/PlayerCountry.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (moizarafat) [inline src/main/java/example/models/PlayerCountry.java:1]: we don't need this entity cos the table is anyways created through changelog.xml

File: src/main/java/example/models/PlayerStats.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (moizarafat) [inline src/main/java/example/models/PlayerStats.java:1]: We dont need this entity/class cos its created through changelog.xml

File: src/main/resources/models/security.hjson [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (moizarafat) [inline src/main/resources/models/security.hjson:1]: Lets remove this file for now based on the comment in elide-standalone. We dont support rules completely yet and the roles we are creating here are not being used in the Tables.hjson.

File: src/main/resources/models/tables/Player.hjson [hunk 6]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (moizarafat) [inline src/main/resources/models/tables/Player.hjson:1]: Lets remove this file as its quite similar to playerStatsView. Based on one of the earlier comments, we should keep the example code repo simple.

File: src/main/resources/models/tables/PlayerCountryView.hjson [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+    tables: [{
+        name: PlayerCountryView
```
Comment (moizarafat) [inline src/main/resources/models/tables/PlayerCountryView.hjson:3]: Lets change the name to PlayerCountry here and in the file name to match the table name itself.

File: src/main/resources/models/tables/PlayerStatsView.hjson [hunk 8]
```diff
@@ -0,0 +1,56 @@
+{
+    tables: [{
+        name: PlayerStatsView
```
Comment (moizarafat) [inline src/main/resources/models/tables/PlayerStatsView.hjson:3]: Lets change the name to PlayerStats here and in the file name to match the table name itself.

File: src/test/java/example/ExampleTest.java [hunk 9]
```diff
@@ -75,6 +75,30 @@ void jsonApiTest() {
...
+    }
+    
+    /**
+     * This test dynamic config created view
+     */
+    @Test
+    void playerStatsViewTest() {
+    	given()
+        .when()
+        .get("/api/v1/playerStatsView")
```
Comment (moizarafat) [inline src/test/java/example/ExampleTest.java:97]: This test will not be needed based on the other review comments.

File: pom.xml [hunk 10]
```diff
@@ -33,6 +33,16 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
```
Comment (moizarafat) [inline pom.xml:36]: I think we don't need to include any of these dependencies here. cos elide-standalone dependecy will pull everything. Please remove and verify,

File: src/main/resources/db/changelog/changelog.xml [hunk 11]
```diff
@@ -25,6 +25,18 @@
 	  PRIMARY KEY (name),
 	  FOREIGN KEY (artifact_name) REFERENCES ArtifactProduct(name)
 	);
+	CREATE TABLE PlayerStats (
+	  name varchar(255) NOT NULL,
+      countryId varchar(255) DEFAULT NULL,
```
Comment (moizarafat) [inline src/main/resources/db/changelog/changelog.xml:30]: indentation is off.

File: src/main/resources/db/changelog/changelog.xml [hunk 12]
```diff
@@ -25,6 +25,18 @@
...
+	  highScore integer DEFAULT NULL,
+	  createdOn date,
+	  PRIMARY KEY (name)
+	);
+	CREATE TABLE PlayerCountry (
+	  id varchar(255) NOT NULL,
+      isoCode varchar(255) DEFAULT NULL,
```
Comment (moizarafat) [inline src/main/resources/db/changelog/changelog.xml:37]: indentation is off.

File: src/test/java/example/ExampleTest.java [hunk 13]
```diff
@@ -75,6 +75,18 @@ void jsonApiTest() {
                 .statusCode(HttpStatus.SC_OK);
     }
 
+    /**
+     * This test static model
+     */
+    @Test
+    void playerStatsTest() {
+    	given()
```
Comment (moizarafat) [inline src/test/java/example/ExampleTest.java:83]: The indentation looks off here. Probably there is a tab in one of the lines.

File: src/main/resources/models/tables/PlayerStats.hjson [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                type: toOne
+                definition: '%join.id = %from.countryId'
+            }
+        ]
+        measures : [
+            {
+            name : highScore
+            type : INTEGER
+            definition: 'MAX(highScore)'
+            readAccess : Prefab.Role.All
```
Comment (moizarafat) [inline src/main/resources/models/tables/PlayerStats.hjson:24]: we don't need to pass readAccess for every column if its same as the readAccess at table level.

File: src/main/resources/models/tables/PlayerCountry.hjson [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+        '''
+        A long description
+        '''
+        cardinality : small
+        readAccess : Prefab.Role.All
+        dimensions : [
+            {
+                name : isoCode
+                type : TEXT
+                readAccess : Prefab.Role.All
```
Comment (moizarafat) [inline src/main/resources/models/tables/PlayerCountry.hjson:15]: we can remove this.

**Comment by aklish** [comment]
I absorbed these changes in another PR.

**Pull request closed without merging**